### PR TITLE
Make provider's name unique per user

### PR DESCRIPTION
**Pull request by nnorbert**
Make provider's name unique per user.

### CODE REVIEW

**Review by peschina — APPROVED** [review]
LGTM, just a few minor comments

File: apps/payments/migrations/sqls/20240507140208-update-provider-name-field-unique-down.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+ALTER TABLE payment_providers
+DROP CONSTRAINT unique_provider_name;
+
+COMMIT;
```
[the remark this answers is not in the document]
Comment (nnorbert) [reply apps/payments/migrations/sqls/20240507140208-update-provider-name-field-unique-down.sql]: Done

File: apps/payments/migrations/sqls/20240507140208-update-provider-name-field-unique-up.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+ALTER TABLE payment_providers
+ADD CONSTRAINT unique_provider_name UNIQUE (user_id, provider_name);
+
+COMMIT;
```
[the remark this answers is not in the document]
Comment (nnorbert) [reply apps/payments/migrations/sqls/20240507140208-update-provider-name-field-unique-up.sql]: Done

File: apps/payments-api/routes/providers/index.ts [hunk 3]
```diff
@@ -13,8 +13,47 @@ import {
...
 
 export default async function providers(app: FastifyInstance) {
+  const dbConstrainMap: Record<string, { field: string; message: string }> = {
+    unique_provider_name: {
+      field: "providerName",
+      message: "Provider's name must be unique!",
+    },
+  };
+
+  const handleDbError = (err: unknown) => {
```
Comment (alfonsograziano) [inline apps/payments-api/routes/providers/index.ts:31]: Shall this function live in a utils file?

**Review by alfonsograziano — APPROVED** [review]
LGTM - just left an optional comment

File: apps/payments-api/routes/providers/index.ts (hunk 3 above)
Comment (nnorbert) [reply apps/payments-api/routes/providers/index.ts:31]: Maybe yes. Let me move this.

Comment (nnorbert) [reply apps/payments-api/routes/providers/index.ts]: Done

**Pull request merged**